packages/peregrine/lib/talons/OrderHistoryPage/useOrderRow.js

```diff
@@ -29,17 +32,49 @@ export const useOrderRow = props => {
             urlKeys
         }
     });
+
+    const { data: getConfigurableThumbnailSourceData } = useQuery(
+        getConfigurableThumbnailSource,
+        {
+            fetchPolicy: 'cache-and-network'
+        }
+    );
+
+    const configurableThumbnailSource = useMemo(() => {
+        if (getConfigurableThumbnailSourceData) {
+            return getConfigurableThumbnailSourceData.storeConfig
+                .configurable_thumbnail_source;
+        }
+    }, [getConfigurableThumbnailSourceData]);
+
     const imagesData = useMemo(() => {
         if (data) {
-            // filter out items returned that we didn't query for
-            const filteredItems = data.products.items.filter(item =>
-                urlKeys.includes(item.url_key)
-            );
-            return filteredItems;
+            // Images data is taken from simple product or from configured variant and assigned to item sku
+            const mappedImagesData = {};
+            items.forEach(item => {
+                const product = data.products.items.find(
+                    element => item.product_url_key === element.url_key
+                );
+                if (
+                    configurableThumbnailSource === 'itself' &&
+                    product.variants &&
+                    product.variants.length > 0
+                ) {
+                    const foundVariant = product.variants.find(variant => {
+                        return variant.product.sku === item.product_sku;
+                    });
+                    mappedImagesData[item.product_sku] =
+                        foundVariant && foundVariant.product;
```

Comment:
What happens if `foundVariant` is falsy?

Reply:
Good question actually it should not be falsy if we got correct data for configurable product in the order with variants, so one of the variants should have the same SKU as order item.